client/engine/messageservice/p2p-message-service/service.go

```diff
@@ -164,6 +242,20 @@ func (s *P2PMessageService) Out() <-chan protocols.Message {
 
 // Close closes the P2PMessageService
 func (s *P2PMessageService) Close() error {
+	s.cancel()
+	s.mdns.Close()
 	s.p2pHost.RemoveStreamHandler(PROTOCOL_ID)
 	return s.p2pHost.Close()
 }
+
+// PeerInfoReceived returns a channel that receives a PeerInfo when a peer is discovered
+func (s *P2PMessageService) PeerInfoReceived() <-chan PeerInfo {
+	return s.newPeerInfo
+}
+
+// AddPeers is DEFUNCT and is only here for testground compatibility
+func (ms *P2PMessageService) AddPeers(peers []PeerInfo) {
```

Comment:
This can be removed once https://github.com/statechannels/go-nitro-testground/pull/161 is merged